src/browser/components/MattermostView.jsx

```diff
@@ -97,6 +97,8 @@ export default class MattermostView extends React.Component {
         } else if (Utils.isTeamUrl(this.props.src, e.url, true) || Utils.isPluginUrl(this.props.src, e.url)) {
           // New window should disable nodeIntegration.
           window.open(e.url, remote.app.name, 'nodeIntegration=no, contextIsolation=yes, show=yes');
+        } else if (Utils.isTrustedRemoteUrl(this.props.src, e.url)) {
+          e.preventDefault();
```

Comment:
Why is the code path here different from the plugin URL flow above?

Reply:
Personally I don't like too big if statements..that was the main reason. Another reason is this part of code should not be executed if so, it again triggers 'new-window' which is okay, it will not start new popup because it is started..If you want me I can combine them in one single if

Reply:
This is more of a code quality related thing, so I'll leave the decision to @Willyfrog

Reply:
hum... wouldn't that also happen to plugins? I'll make a note to review that in the future. for now I'm ok with it if we prevent extra events that are not needed.

Reply:
> hum... wouldn't that also happen to plugins? I'll make a note to review that in the future. for now I'm ok with it if we prevent extra events that are not needed.

Plugins will trigger again window.open which will be caught in ```main.js``` line 525:
```  contents.on('new-window', (event, url) => {```

but prevented from opening in 544:
```      if (!popupWindow || popupWindow.closed) {```